app/assets/javascripts/discourse/app/controllers/preferences/profile.js

```diff
@@ -81,6 +93,14 @@ export default Controller.extend({
     document.querySelector(".feature-topic-on-profile-btn")?.focus();
   },
 
+  _missingRequiredFields(siteFields, userFields) {
+    return siteFields
```

Comment:
Might actually be better to do this in the `user.js` model, you have access to `Site` from there, and this will help remove the dependency on the controller in the route file for the profile

Reply:
I guess the concern here is mainly having a route reference its controller? I assume this is less conventional since there's no "native" access, need to use `controllerFor`. At the same time I'm a bit hesitant to chuck this in `User` since it's UI logic related to this particular view only, so it 1) creates more indirection, 2) puts UI specific stuff in the view, and 3) adds to the 1.5k LOC already in `User`. 😬

Reply:
Yes that's it, it feels very odd for the route have to dig into controller logic like this (also the fact that so much logic is in the controller since it's an old route is :nauseated_face: ), and also since the condition is `showEnforcedRequiredFieldsNotice` which is mostly about a UI element on the page.

I just realised that is only pointing to `user.needs_required_fields_check` , so we should check `user.needs_required_fields_check` directly in the route, that would work better for me.

Reply:
> I just realised that is only pointing to user.needs_required_fields_check , so we should check user.needs_required_fields_check directly in the route, that would work better for me.

This is it! 🚀